src/EditorFeatures/CSharpTest/UseLocalFunction/UseLocalFunctionTests.cs

```diff
@@ -2495,6 +2520,56 @@ void Goo()
 }");
         }
 
+        [WorkItem(23872, "https://github.com/dotnet/roslyn/issues/23872")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseLocalFunction)]
+        public async Task TestSimpleInitialization_SingleLine4()
```

Comment:
@CyrusNajmabadi  how do I reconcile this with your new set of tests from #24867? I think anon. methods have to be here as well and I added an async version of tests 2 & 4 everywhere. If you want int[] as a return type covered maybe it can be put in existing tests instead of making even more of them?

Reply:
>  how do I reconcile this with your new set of tests

Copy/paste woudl work?  :)   I think it would make sense to just merge this branch into yours.  If there's a conflict with tests, just keep yours and then manually add these few tests in afterwards.

Thanks!

Reply:
@CyrusNajmabadi yes but I added more tests yesterday and some of them do the same thing as your test for async (but not for int[]) and mine are organized differently, I have tests 1, 2, 2Async, 3, 4, 4Async (previously your progression was just for different kinds of lambdas, so I added 4 for anonymous methods and a separate async version of some tests).

if i add yours just the way they are it's going to be a mess with duplicates

Reply:
If you feel like your tests cover the same set of cases that mine do, you can just remove my tests and use yours.  If you feel like your tests don't get the same coverage (for example, they don't test multi-token return types), then i would keep my tests (but organize/augment them however you want).

Reply:
i.e. you can rename them, order them anywhere, even edit them.  The core goal of these tests was to just ensure multi-token return types (in either async or sync contexts) don't cause wrapping.

Reply:
I was asking if it would be OK to have a multi-token return type in every test

Reply:
mine have to cover anonymous methods with async as well of course, so one option would be to add another set of tests with AsyncMultitoken or something like that to both lambdas &anon. methods, or just always use multitoken with async, or with all tests, which would be simpler

Reply:
> I was asking if it would be OK to have a multi-token return type in every test

I'm always fine with more tests :)

It's totally up to you.  I was satisfied with the number of tests i wrote to validate this scenario.  If you think it's not enough, adding more is totally up to you :)

Reply:
no that's not what I meant, I was asking if it would be a bad idea to put int[] on all of them

Reply:
otherwise i'm going to have to add 2(non async multitoken + async multitoken) * 2(lambdas+anon.methods) * 3 more tests, so that's 12

Reply:
>  I was asking if it would be a bad idea to put int[] on all of them

Reply:
> otherwise i'm going to have to add 2(non async multitoken + async multitoken) * 2(lambdas+anon.methods) * 3 more tests, so that's 12... there's already 24 tests for this single line issue

Again, this is up to you.  I'm satisfied with the current level of testing.  If you are not, you are perfectly free to add more tests.  I'm not going to stop you.  :)

Reply:
not more variants, only int[] instead of int everywhere... otherwise i'll have to add 12 more tests

Reply:
> not more variants, only int[] instead of int everywhere... 

i would be fine with you using int[] everywhere (though at least one single-token return type test would be good then too).

> otherwise i'll have to add 12 more tests to cover these variations for anon methods as well as lambdas

Sorry, i'm not getting the "have to add" part.  As i mentioned already, i think the current level of coverage for this case is fine.  I'm not asking you to add all these tests.  So i'm not sure why you think you would "have" to do it :)

BTW, i'm in gitter if you'd like to discuss things there.  It might make things go a lot faster :)